qsimcirq/qsim_circuit.py

```diff
@@ -34,103 +34,182 @@
+        return qsim.kH
+    return qsim.kHPowGate
+
+
+def _translate(gate: cirq.CZPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kCZ
+    return qsim.kCZPowGate
+
+
+def _translate_CXPowGate(gate: cirq.CXPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kCX
+    return qsim.kCXPowGate
+
+
+def _translate_PhasedXPowGate(gate: cirq.PhasedXPowGate):
+    return qsim.kPhasedXPowGate
+
+
+def _translate_PhasedXZGate(gate: cirq.PhasedXZGate):
+    return qsim.kPhasedXZGate
+
+
+def _translate_XXPowGate(gate: cirq.XXPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kXX
+    return qsim.kXXPowGate
+
+
+def _translate_YYPowGate(gate: cirq.YYPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kYY
+    return qsim.kYYPowGate
+
+
+def _translate_ZZPowGate(gate: cirq.ZZPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kZZ
+    return qsim.kZZPowGate
+
+
+def _translate_SwapPowGate(gate: cirq.SwapPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kSWAP
+    return qsim.kSwapPowGate
+
+
+def _translate_ISwapPowGate(gate: cirq.ISwapPowGate):
+    # cirq.riswap also uses this path.
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kISWAP
+    return qsim.kISwapPowGate
+
+
+def _translate_PhasedISwapPowGate(gate: cirq.PhasedISwapPowGate):
+    # cirq.givens also uses this path.
+    return qsim.kPhasedISwapPowGate
+
+
+def _translate_FSimGate(gate: cirq.FSimGate):
+    return qsim.kFSimGate
+
+
+def _translate_TwoQubitDiagonalGate(gate: cirq.TwoQubitDiagonalGate):
+    return qsim.kTwoQubitDiagonalGate
+
+
+def _translate_ThreeQubitDiagonalGate(gate: cirq.ThreeQubitDiagonalGate):
+    return qsim.kThreeQubitDiagonalGate
+
+
+def _translate_CCZPowGate(gate: cirq.CCZPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kCCZ
+    return qsim.kCCZPowGate
+
+
+def _translate_CCXPowGate(gate: cirq.CCXPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+        return qsim.kCCX
+    return qsim.kCCXPowGate
+
+
+def _translate_CSwapGate(gate: cirq.CSwapGate):
+    return qsim.kCSwapGate
+
+
+def _translate_MatrixGate(gate: cirq.MatrixGate):
+    if gate.num_qubits() <= 6:
+        return qsim.kMatrixGate
+    raise NotImplementedError(
+        f"Received matrix on {gate.num_qubits()} qubits; "
+        + "only up to 6-qubit gates are supported."
+    )
+
+
+def _translate_MeasurementGate(gate: cirq.MeasurementGate):
+    # needed to inherit SimulatesSamples in sims
+    return qsim.kMeasurement
+
+
+TYPE_TRANSLATOR = {
+    cirq.ControlledGate: _translate_ControlledGate,
+    cirq.XPowGate: _translate_XPowGate,
+    cirq.YPowGate: _translate_YPowGate,
+    cirq.ZPowGate: _translate_ZPowGate,
+    cirq.HPowGate: _translate_HPowGate,
+    cirq.CXPowGate: _translate_CXPowGate,
```

Comment:
It seems `CZPowGate` is missing here.

Reply:
Added (likely a copy-paste error) with negligible effect on benchmark results.